### PR TITLE
Ling accent, skin tone, hair gradient, and organ failure fixes

### LINKED ISSUE #18338: [Bug]: Changeling emergency transform gave me the wrong organs
### Checks

- [x] I have searched the bug with a few keywords, and I confirm this bug was not yet reported.

### Location

Horizon

### Describe the issue

So basically as ling I used emergency transform, and when I was able to regrow my body it was all kinds of messed up. 

- Deathly pale (normally tan)
- Wrong accent
- Organ failure

I think the first one and maybe second one are more common, I'm not exactly sure how I ended up with organ failure. It might have something to do with that character having invalid choices that get reset when I load them.

Bug occurred in round before cr4-a7pW

### How to reproduce

1. Be ling
2. Emergency transform
3. Regrow body
4. ???
5. Die of total organ failure

### Round ID

_No response_

### Staff ckey

_No response_

### PR DESCRIPTION
**Pull request by AlaunusLux**
Reworked emergency transform to revert to your chosen form at the time of ejection, fixed the accent and skin tone changing, as well as organ failure. 
Separately, added a datum to track gradients because absorbDNA's proc call was getting too long. Maybe get_gradient() would be better.

Pretty concerned about ripple effects due to the change in the dna file. I will look into it more later, but I am assuming that either this function isn't used in login or character creation, or the value gets skewed somehow as ling. Will look into a different fix, but hoped someone could shed some light on that.

Fixes #18338

### CODE REVIEW

**Comment by AlaunusLux** [comment]
!bugfix

**Comment by AlaunusLux** [comment]
!review

File: code/game/dna/dna2_helpers.dm [hunk 1]
```diff
@@ -145,7 +145,7 @@
 		H.b_eyes   = dna.GetUIValueRange(DNA_UI_EYES_B,    255)
 		H.update_eyes()
 
-		H.s_tone   = 35 - dna.GetUIValueRange(DNA_UI_SKIN_TONE, 220) // Value can be negative.
+		H.s_tone   = dna.GetUIValueRange(DNA_UI_SKIN_TONE, 220)
```
Comment (AlaunusLux) [inline code/game/dna/dna2_helpers.dm:148]: What's up with this?

**Comment by AlaunusLux** [comment]
Wanted to clarify my thought process on turning the ling to whatever form they had most recently, instead of original/what they spawned as. Lorewise I wouldn't consider lings to have an original humanoid form. Either they replaced your character or your character was a ling waiting to perhaps get more DNA or a new body or whatever. It could be played as some awakening of abilities or something, but generally with the thing inspiration, it doesn't seem like it would be attached to any form over another.

Mechanically, I would imagine you're in the form you want to be in before decoupling your head - due to the genetic damage when regrowing a body, it takes a while to transform into who I assume you'd want to be, and that's after waiting five minutes in squishy head form. If this is intentional and wanted then it can be reversed of course.

File: code/game/dna/dna2_helpers.dm (hunk 1 above)
Comment (NonQueueingMatt) [reply code/game/dna/dna2_helpers.dm:148]: Back in the ancient era, skin tone was calculated by doing `35 - x`, where `x` is a value from 30 to 220.

**Review by NonQueueingMatt — APPROVED** [review]

**Pull request merged**